src/khoj/processor/notion/notion_to_jsonl.py

```diff
@@ -0,0 +1,126 @@
+# Standard Packages
+import logging
+
+# External Packages
+import requests
+
+# Internal Packages
+from khoj.utils.helpers import timer
+from khoj.utils.rawconfig import Entry, NotionContentConfig
+from khoj.processor.text_to_jsonl import TextToJsonl
+from khoj.utils.jsonl import dump_jsonl, compress_jsonl_data
+from khoj.utils.rawconfig import Entry
+
+
+logger = logging.getLogger(__name__)
+
+
+class NotionToJsonl(TextToJsonl):
+    def __init__(self, config: NotionContentConfig):
+        super().__init__(config)
+        self.config = config
+        self.session = requests.Session()
+        self.session.headers.update({"Authorization": f"Bearer {config.token}", "Notion-Version": "2022-02-22"})
+
+    def process(self, previous_entries=None):
+        current_entries = []
+
+        # Get all pages
+        with timer("Getting all pages via search endpoint", logger=logger):
+            responses = []
+
+            while True:
+                result = self.session.post(
+                    "https://api.notion.com/v1/search",
+                    json={"page_size": 100},
+                ).json()
+                responses.append(result)
+                if result["has_more"] == False:
+                    break
+                else:
+                    self.session.params = {"start_cursor": responses[-1]["next_cursor"]}
+
+        for response in responses:
+            with timer("Processing response", logger=logger):
+                pages_or_databases = response["results"]
+
+                # Get all pages content
+                for p_or_d in pages_or_databases:
+                    with timer(f"Processing {p_or_d['object']} {p_or_d['id']}", logger=logger):
+                        if p_or_d["object"] == "database":
+                            # TODO: Handle databases
+                            continue
+                        elif p_or_d["object"] == "page":
+                            page_id = p_or_d["id"]
+                            title, content = self.get_page_content(page_id)
+                            raw_content = ""
+                            curr_heading = ""
+                            for block in content["results"]:
+                                block_type = block.get("type", None)
+                                if block_type == None:
+                                    continue
+                                block_data = block[block_type]
+                                raw_content = ""
+                                if block_data.get("rich_text", None) and len(block_data["rich_text"]) > 0:
```

Comment:
Is the default to None required here or is something like below fine?
```suggestion
                                if block_data.get("rich_text") and len(block_data["rich_text"]) > 0:
```